pkg/dao/repository_config.go

```diff
@@ -489,3 +481,70 @@ func (r repositoryConfigDaoImpl) ValidateUrl(orgId string, url string, response
 	}
 	return nil
 }
+
+func (r repositoryConfigDaoImpl) ValidateUrlPresence(url string, response *api.RepositoryValidationResponse) *string {
+	repomd, code, err := r.extResDao.FetchRepoMd(url)
+	if err != nil {
+		response.URL.HTTPCode = code
+		if isTimeout(err) {
+			response.URL.Error = fmt.Sprintf("Error fetching YUM metadata: %s", "Timeout occurred")
+		} else {
+			response.URL.Error = fmt.Sprintf("Error fetching YUM metadata: %s", err.Error())
+		}
+		response.URL.MetadataPresent = false
+		return nil
+	} else {
+		response.URL.HTTPCode = code
+		response.URL.MetadataPresent = code >= 200 && code < 300
+		return repomd
+	}
+}
+
+func (r repositoryConfigDaoImpl) checkSignaturePresent(request *api.RepositoryValidationRequest, repomd *string, response *api.RepositoryValidationResponse) {
+	if request.GPGKey == nil || *request.GPGKey == "" {
+		response.GPGKey.Skipped = true
+		response.GPGKey.Valid = true
+	} else {
+		_, err := LoadGpgKey(request.GPGKey)
+		if err == nil {
+			response.GPGKey.Valid = true
+		} else {
+			response.GPGKey.Valid = false
+			response.GPGKey.Error = fmt.Sprintf("Error loading GPG Key: %e.  Is this a valid GPG Key?", err)
```

Comment:
```suggestion
			response.GPGKey.Error = fmt.Sprintf("Error loading GPG Key: %w.  Is this a valid GPG Key?", err)
```
I think you meant to use %w for error here. %e is for scientific notation.

Reply:
yes!

Reply:
although that didn't actually work, switched to %s